android/app/src/main/java/app/organicmaps/routing/RoutingController.java

```diff
@@ -351,7 +351,7 @@ public void prepare(@Nullable MapObject startPoint, @Nullable MapObject endPoint
   private void initLastRouteType(@Nullable MapObject startPoint, @Nullable MapObject endPoint,
                                  boolean fromApi)
   {
-    if (isSubwayEnabled() && !fromApi)
+    if (isInactive() && isSubwayEnabled() && !fromApi)
```

Comment:
What does `isInactive` mean in the context of this controller? Is it the best name?

Reply:
I struggled a bit to find a better name, but couldn't really come up with anything satisfying. The controller being inactive means you're not planning nor navigating. Maybe invert it and call it `isInUse`?

Reply:
WDYT about moving `!fromApi` and mState check into `isSubwayEnabled()` below, and rename it to something like `shouldForceTransitRoute`?

Reply:
Mmm, that does sound alright, `isSubwayEnabled` isn't used anywhere else. I'll push the change in a sec